Dockerfile

```diff
@@ -1,50 +1,41 @@
-FROM python:3.5
-MAINTAINER Pit Kleyersburg <pitkley@googlemail.com>
-
-# Install dependencies
-RUN apt-get update \
-    && apt-get install -y --no-install-recommends \
-        sudo \
-        tesseract-ocr tesseract-ocr-eng imagemagick ghostscript unpaper \
-    && rm -rf /var/lib/apt/lists/*
-
-# Install python dependencies
-RUN mkdir -p /usr/src/paperless
+FROM alpine:3.7
 WORKDIR /usr/src/paperless
 COPY requirements.txt /usr/src/paperless/
-RUN pip install --no-cache-dir -r requirements.txt
-
 # Copy application
-RUN mkdir -p /usr/src/paperless/src
-RUN mkdir -p /usr/src/paperless/data
-RUN mkdir -p /usr/src/paperless/media
 COPY src/ /usr/src/paperless/src/
 COPY data/ /usr/src/paperless/data/
 COPY media/ /usr/src/paperless/media/
-
+# Set export directory
+ENV PAPERLESS_EXPORT_DIR /export
 # Set consumption directory
 ENV PAPERLESS_CONSUMPTION_DIR /consume
-RUN mkdir -p $PAPERLESS_CONSUMPTION_DIR
-
+COPY scripts/docker-entrypoint.sh /sbin/docker-entrypoint.sh
+# Install dependencies
+RUN apk --no-cache --update add \
+        python3 gnupg libmagic bash \
+        sudo tesseract-ocr imagemagick ghostscript unpaper && \
+    apk --no-cache add --virtual .build-dependencies \
+        python3-dev gcc musl-dev zlib-dev jpeg-dev && \
+## Install python dependencies
+    python3 -m ensurepip && \
+    rm -r /usr/lib/python*/ensurepip && \
+    mkdir -p /usr/src/paperless && \
+    pip3 install --no-cache-dir -r requirements.txt && \
```

Comment:
In reference to the other comment regarding the removal of the one `WORKDIR` instruction, it probably is enough to add a `cd /usr/src/paperless` between the `mkdir` and `pip3` commands.

The latter `WORKDIR` instruction ensures that we execute from the right directory.